Patch-Source: https://gitlab.alpinelinux.org/alpine/abuild/-/merge_requests/438
---
From e06205363314812ff5993c154aa127f5fafa8faa Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Tue, 30 Sep 2025 14:42:58 +0200
Subject: [PATCH] default.conf: use libcxx's -D_LIBCPP_HARDENING_MODE instead
 of -D_LIBCPP_ENABLE_HARDENED_MODE

LLVM 21 removed the deprecated -D_LIBCPP_ENABLE_HARDENED_MODE option in
favor of -D_LIBCPP_HARDENING_MODE, so use that one instead with the
"fast" mode, see https://libcxx.llvm.org/Hardening.html
---
 default.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/default.conf b/default.conf
index e4abe6b0..33290ae8 100644
--- a/default.conf
+++ b/default.conf
@@ -1,6 +1,6 @@
 export CFLAGS="-Os -fstack-clash-protection -Wformat -Werror=format-security"
 export CXXFLAGS="-Os -fstack-clash-protection -Wformat -Werror=format-security"
-export CXXFLAGS="$CXXFLAGS -D_GLIBCXX_ASSERTIONS=1 -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS=1 -D_LIBCPP_ENABLE_HARDENED_MODE=1"
+export CXXFLAGS="$CXXFLAGS -D_GLIBCXX_ASSERTIONS=1 -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS=1 -D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST"
 export LDFLAGS="-Wl,--as-needed,-O1,--sort-common"
 export GOFLAGS="-buildmode=pie -modcacherw -trimpath -buildvcs=false"
 # Do note that these should work with at least GDC and LDC
-- 
GitLab

